integration-tests/pkgJson-restore.spec.js

```diff
@@ -14,1449 +14,681 @@
+                    name: PLUGIN_ID,
+                    spec: `git+${PLUGIN_URL}.git`,
+                    variables: jasmine.any(Object)
+                }]);
+                // Plugin was removed from the installed plugin list successfully.
+                expect(pluginPath(PLUGIN_ID)).not.toExist();
+            }).then(function () {
+                // Add platform (so that prepare can run).
+                return cordovaPlatform('add', 'browser', {save: true});
+            }).then(function () {
+                return prepare({save: true});
+            }).then(function () {
+                // Config.xml spec now matches the one in pkg.json.
+                expectConsistentPlugins([{
+                    name: PLUGIN_ID,
+                    spec: `git+${PLUGIN_URL}.git`,
+                    variables: jasmine.any(Object)
+                }]);
+            });
         });
-        var configEngArray = engNames.slice();
 
-        // Expect that pkg.json exists without a cordova key.
-        expect(pkgJson).toBeDefined();
-        expect(pkgJson.cordova).toBeUndefined();
-        // Expect that config.xml contains only android at this point.
-        expect(configEngArray.indexOf('android')).toBeGreaterThan(-1);
-        expect(configEngArray.length === 1);
-
-        return emptyPlatformList().then(function () {
-            return prepare();
-        }).then(function () {
-            var cfg2 = new ConfigParser(configXmlPath);
-            engines = cfg2.getEngines();
-            engNames = engines.map(function (elem) {
-                return elem.name;
+        /** Test#003 will add two platforms to package.json - one with the 'save' flag and one
+        *   without the 'save' flag. It will remove both platforms without a 'save' flag.
+        *   After running cordova prepare, only the platform added with the 'save' flag is restored
+        *   in platforms.json.
+        */
+        it('Test#003 : should NOT restore platform that was not saved and removed', function () {
+            const PLATFORM_1 = 'ios';
+            const PLATFORM_2 = 'browser';
+
+            expect(installedPlatforms()).toEqual([]);
+
+            return Promise.resolve().then(function () {
+                // Add PLATFORM_1 platform to project without --save
+                return cordovaPlatform('add', PLATFORM_1);
+            }).then(function () {
+                // Add PLATFORM_2 to project with --save
+                return cordovaPlatform('add', PLATFORM_2, {save: true});
+            }).then(function () {
+                // Both platforms are installed but only PLATFORM_2 is in pkg.json
+                expect(installedPlatforms()).toEqual([PLATFORM_2, PLATFORM_1]);
+                expect(getPkgJson('cordova.platforms')).toEqual([PLATFORM_2]);
+            }).then(function () {
+                // Remove all platforms without --save.
+                return cordovaPlatform('rm', [PLATFORM_1, PLATFORM_2]);
```

Comment:
Looks like a very minor change in tested behavior from the following:
```js
                // Remove helpers.testPlatform without --save.
                return cordovaPlatform('rm', ['browser']);
```

I would favor avoiding change in tested behavior during this kind of cleanup commit if possible.

Reply:
Yes, I changed it slightly so the test makes actual sense and tests what the description says. You're right that this should go into a separate PR though. I have some similar changes prepared anyway.

Reply:
I only joined the two adjacent calls to `cordovaPlatform('rm', ...);` into one. I wouldn't say that this changes tested behavior. Especially since `restore` is under test here, not `cordovaPlatform`.